### PR TITLE
Coverage:  Add `calculate_coverage` function.

### PR DESCRIPTION
**Pull request by kindly**
This is a pull requests for #35 

@duncandewhurst could you look over the documentation and have a go with the following colab sheet:

https://colab.research.google.com/drive/14B0uxBoyFRScCGb7xnBK_fNuKHjGZ_U4

It loads in some sample/random data that we use for testing kinglisher summaraize and contains most fields in the standard.  You could also use some other real data but you will have to import this branch using the pip install line in the first line in the notebook and import the calculate_coverage function.

### CODE REVIEW

**Review by duncandewhurst — CHANGES_REQUESTED** [review]
Thanks, @kindly!

I've pinged you some comments in the notebook and made some suggestions for the documentation.

File: ocdskingfishercolab/__init__.py [hunk 1]
```diff
@@ -365,6 +365,145 @@ def render_json(json_string):
         """)
 
 
+def calculate_coverage(fields, scope=None, sql=True):
+    """
+    Calculate the coverage of the fields supplied.
+
+    `scope` is the table name of where the coverage is going to be counted from;
+    the total rows in this table will be the denomitor of the coverage percentage.
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply ocdskingfishercolab/__init__.py:375]: @kindly what do you think?

File: ocdskingfishercolab/__init__.py [hunk 2]
```diff
@@ -365,6 +366,175 @@ def render_json(json_string):
...
+
+    `fields` is a list of fields to measure the coverage of, specified using JSON Pointer relative
+    to the object summarized in the scope table; e.g. to measure the coverage of `awards/value/amount`
+    against `awards_summary`, set `scope` to `awards` and `fields` to `awards/value/amount`.
+
+    To specify fields that are children of the scope table, you can prefix the pointer with a `:`
+    e.g. for the above example you could set the `scope` table to `awards` and `fields` to `:value/amount`.
```
Comment (duncandewhurst) [inline ocdskingfishercolab/__init__.py:382]: Update to reflect new notation/logic.

```suggestion
    `fields` is a list of fields to measure the coverage of, specified using JSON Pointer.
    
    To specify fields that are children of the scope table, you can use either an absolute pointer or a relative pointer prefixed with `:`, e.g. if `scope` is set to 'awards', then `awards/value/amount` and `:value/amount` refer to the same field. Coverage of such fields is measured against the number of rows in the `scope` table.
    
    To specify fields that are not children of the scope table, use an absolute path, e.g. `tender/procurementMethod`. Coverage of such fields is measured against the number of releases/records.
```

File: ocdskingfishercolab/__init__.py [hunk 3]
```diff
@@ -365,6 +366,175 @@ def render_json(json_string):
...
+
+   To specify that a field must appear in **all** objects in the array, prepend the field with `ALL `,
+   e.g. if `scope` is set to `awards` and `field` is set to `ALL items/description`,
+   all items must have a description for the coverage to be non-zero.
+
+    To measure the co-occurence of fields across awards and contracts related by `contracts/awardID`,
+    set `scope` to `awards` and prepend `:contracts/` to the path of fields in contracts, or set `scope` to `contracts`
+    and prepend `:awards/` to the path of fields in awards;
+    e.g. to measure how many contracts have a value and a related award with suppliers,
+    set `scope` to `contracts` and `fields` to `[':value', ':awards/suppliers/id']`.
```
Comment (duncandewhurst) [inline ocdskingfishercolab/__init__.py:396]: Simplify

```suggestion
    If `scope` is set to `awards`, specify fields on related contracts by prefixing the path with `:contracts/`, e.g. to measure how many awards have a value and a related contract with a period, set `scope` to `awards` and `fields` to `[':value', ':contracts/period']`. Similarly, if `scope` is set to `contracts`, specify fields on related awards by prefixing the path with `:awards/`.
```

File: ocdskingfishercolab/__init__.py [hunk 4]
```diff
@@ -365,6 +366,175 @@ def render_json(json_string):
...
+
+    If the field is not prepended with a `:` it will look at its related release/record as the base for the path.
+    So for example if you are looking at `contracts` and want to check for any or all `awards`
+    *on the same release/record* has a value you can specify `awards/value`.
+
+    A total coverage percentage of all the fields co-occuring together is also calculated.
```
Comment (duncandewhurst) [inline ocdskingfishercolab/__init__.py:402]: Remove as covered above.

```suggestion
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]
Thanks @kindly I've suggested docs updates to reflect the new notation/logic and to improve the flow. There's one comment outstanding from my previous review: https://github.com/open-contracting/kingfisher-colab/pull/55#discussion_r780867708

File: ocdskingfishercolab/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kindly) [reply ocdskingfishercolab/__init__.py:375]: I have changed it to the table name.

**Pull request merged**